From b02d7dc690f2acdc73acdb847225c974089d2a90 Mon Sep 17 00:00:00 2001
From: Matt Porter <mporter@ti.com>
Date: Fri, 7 Sep 2012 15:14:08 -0400
Subject: [PATCH 11/11] spi: mcspi: add hack to work around a bug in the edma
 dmaengine driver

The slave config handling in the edma dmaengine driver is expecting
the direction to be set in order to determine which set of parameters
to load into the channel config. This needs to be addressed in the
edma driver but this hack will do for the moment.

Signed-off-by: Matt Porter <mporter@ti.com>
---
 drivers/spi/spi-omap2-mcspi.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/spi/spi-omap2-mcspi.c b/drivers/spi/spi-omap2-mcspi.c
index d6c5e99..1e8871d 100644
--- a/drivers/spi/spi-omap2-mcspi.c
+++ b/drivers/spi/spi-omap2-mcspi.c
@@ -372,6 +372,7 @@ omap2_mcspi_txrx_dma(struct spi_device *spi, struct spi_transfer *xfer)
 		struct dma_async_tx_descriptor *tx;
 		struct scatterlist sg;
 
+		cfg.direction = DMA_MEM_TO_DEV;
 		dmaengine_slave_config(mcspi_dma->dma_tx, &cfg);
 
 		sg_init_table(&sg, 1);
@@ -394,6 +395,7 @@ omap2_mcspi_txrx_dma(struct spi_device *spi, struct spi_transfer *xfer)
 		struct scatterlist sg;
 		size_t len = xfer->len - es;
 
+		cfg.direction = DMA_DEV_TO_MEM;
 		dmaengine_slave_config(mcspi_dma->dma_rx, &cfg);
 
 		if (l & OMAP2_MCSPI_CHCONF_TURBO)
-- 
1.7.7.6

